docs/explanation/cryptography.rst

```diff
@@ -0,0 +1,303 @@
+    - ``flutter``
+    - Git
+
+  * - :ref:`Matter SDK <reference_matter_sdk_plugin>`
+    - ``matter``
+    - Git
+
+core22
+------
+
+Snaps using core22 as their base snap depend on various craft libraries to build.
+Snapcraft's cryptographic functionality with these libraries is described below.
+
+Public key signing
+~~~~~~~~~~~~~~~~~~
+
+Snapcraft supports the adding and verification of arbitrary package repositories. For
+more information, see the `Craft Archives cryptography`_ documentation.
+
+The parts system
+~~~~~~~~~~~~~~~~
+
+Snapcraft makes use of *parts* in project files for declarative builds. Parts specified
+by the user may download arbitrary files, install packages, and more. For more
+information, see the `Craft Parts cryptography`_ documentation.
+
+Creating virtual build environments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Snapcraft instantiates and executes builds on self-allocated virtual instances. For more
+information, see `Craft Providers cryptography`_ documentation.
+
+Downloading build dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Snaps using core22 have the same plugin support as core24. See the core24 section above
+for more information on how Snapcraft downloads and verifies build dependencies.
+
+core20
+------
+
+Snaps using core20 as their base snap depend on various craft libraries to build.
+Snapcraft's cryptographic functionality with these libraries is described below.
+
+Creating virtual build environments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Snapcraft instantiates and executes builds on self-allocated virtual instances. It uses
+the `Requests`_ library to install Multipass on Windows. Build environments for other
+operating systems handled with snapd, as described below.
+
+Communication with snapd
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Snapcraft uses the Requests library to communicate over Unix sockets with the local
+`snap daemon (snapd)`_. These requests are used to fetch information about required
+software. If the software is missing, Snapcraft will install it through snapd. This
+is done by querying the `snapd API`_ with URLs built dynamically and sanitized by
+`urllib`_.
+
+Sources
+~~~~~~~
+
+Downloading repositories
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+When a part sources a remote repository, Snapcraft uses the appropriate version control
+tool to clone the repository. The protocol used, such as ``SSH`` or ``HTTPS``, depends
+on the source URL and support from the tool.
+
+.. list-table::
+  :header-rows: 1
+
+  * - Version control system
+    - Tool used
+
+  * - `Git`_
+    - ``git``
+
+  * - `Bazaar`_
+    - ``bzr``
+
+  * - `Mercurial`_
+    - ``hg``
+
+  * - `Subversion`_
+    - ``svn```
+
+Downloading source files
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+When a part sources a ``.deb``, ``.rpm``, ``.snap``, ``.tar``, ``.zip``, ``.7z``, or an
+executable file, Snapcraft calls the Requests library to download it.
+
+The integrity of these files can be verified using a :ref:`checksum <source_checksum>`.
```

Comment:
We should describe the circumstances for the verification, even if it's by user selection.

Reply:
Updated https://github.com/canonical/snapcraft/pull/5478/commits/b86011f1e4714e8ec77af1eac46208a99c354aa1